titan-compiler/checker.lua

```diff
@@ -241,17 +241,32 @@ function checkstat(node, st, errors)
         st:add_symbol(node.name, node)
         node._type = node._type or typefromnode(node.type, errors)
     elseif tag == "Stat_Decl" then
-        if node.decl.type then
-          checkstat(node.decl, st, errors)
-          checkexp(node.exp, st, errors, node.decl._type)
-        else
-          checkexp(node.exp, st, errors)
-          node.decl._type = node.exp._type
-          checkstat(node.decl, st, errors)
+        for i = 1, #node.decls do
+            local decl = node.decls[i]
+            local exp = node.exps[i]
```

Comment:
@gligneul I renamed the node attributes to `node.decls` and `node.exps` in plural, so I could rename the locals without causing confusion. (It's good practice to keep those array entries in the AST in plural anyway). Thanks for the nudge in the right direction!

Reply:
:v: